src/main/groovy/com/rundeck/plugins/ansible/plugin/AnsibleResourceModelSource.java

```diff
@@ -345,47 +371,56 @@ public AnsibleRunner.AnsibleRunnerBuilder buildAnsibleRunner() throws ResourceMo
     return runnerBuilder;
   }
 
-
   @Override
   public INodeSet getNodes() throws ResourceModelSourceException {
     NodeSetImpl nodes = new NodeSetImpl();
-    final Gson gson = new Gson();
+    AnsibleRunner.AnsibleRunnerBuilder runnerBuilder = buildAnsibleRunner();
 
+    if (gatherFacts) {
+      processWithGatherFacts(nodes, runnerBuilder);
+    } else {
+      ansibleInventoryList(nodes, runnerBuilder);
+    }
+
+    return nodes;
+  }
+
+  public void processWithGatherFacts(NodeSetImpl nodes, AnsibleRunner.AnsibleRunnerBuilder runnerBuilder) throws ResourceModelSourceException {
+
+    final Gson gson = new Gson();
     Path tempDirectory;
     try {
       tempDirectory = Files.createTempDirectory("ansible-hosts");
     } catch (IOException e) {
-        throw new ResourceModelSourceException("Error creating temporary directory.", e);
+      throw new ResourceModelSourceException("Error creating temporary directory.", e);
     }
 
     try {
-      Files.copy(this.getClass().getClassLoader().getResourceAsStream("host-tpl.j2"), tempDirectory.resolve("host-tpl.j2"));
-      Files.copy(this.getClass().getClassLoader().getResourceAsStream("gather-hosts.yml"), tempDirectory.resolve("gather-hosts.yml"));
+      Files.copy(this.getClass().getClassLoader().getResourceAsStream(HOST_TPL_J2), tempDirectory.resolve(HOST_TPL_J2));
+      Files.copy(this.getClass().getClassLoader().getResourceAsStream(GATHER_HOSTS_YML), tempDirectory.resolve(GATHER_HOSTS_YML));
     } catch (IOException e) {
-        throw new ResourceModelSourceException("Error copying files.");
+      throw new ResourceModelSourceException("Error copying files.");
```

Comment:
Here we are loosing the stack with the reason why the exception occurs

Reply:
Done... ✅